Sources/Fuzzilli/Core/MultiEngine.swift

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/// Wraps multiple engines into one, which can be initialized given a
+/// WeightedList. This can then switch engines to use, or use a more
+/// complicated heuristic.
+public class MultiEngine: ComponentBase, FuzzEngine {
+    let engines: WeightedList<FuzzEngine>
+
+    /// The current active engine.
+    private var activeEngine: FuzzEngine
+
+    /// The number of rounds to complete per engine.
+    private let roundsPerEngine: Int
+
+    /// The number of rounds for the current active engine.
+    private var currentIteration = 0
+
+    public init(engines: WeightedList<FuzzEngine>, initialActive: FuzzEngine? = nil, roundsPerEngine: Int) {
+        assert(roundsPerEngine > 0)
+        self.roundsPerEngine = roundsPerEngine
+        self.engines = engines
+        self.activeEngine = initialActive ?? engines.randomElement()
+        super.init(name: "MultiEngine")
+    }
+
+    override func initialize() {
+        for engine in engines {
+            engine.initialize(with: self.fuzzer)
+        }
+    }
+
+    public func fuzzOne(_ group: DispatchGroup) {
+        activeEngine.fuzzOne(group)
+        currentIteration += 1
+        if currentIteration % roundsPerEngine == 0 {
+            currentIteration = 0
```

Comment:
With the modulo operation in the `if`, you don't need to reset the `currentIteration` variable here I think (Otherwise it should maybe be called `iterationsForActiveEngine` or so instead). If you do that, please update the comment in line 29 as well. Also, please use either `rounds` or `iterations` consistently in this file =)